src/csharp/Microsoft.Spark/Network/DefaultSocketWrapper.cs

```diff
@@ -24,12 +25,20 @@ internal sealed class DefaultSocketWrapper : ISocketWrapper
         /// Default constructor that creates a new instance of DefaultSocket class which represents
         /// a traditional socket (System.Net.Socket.Socket).
         /// 
-        /// This socket is bound to Loopback with port 0.
+        /// This socket is bound to provided IP address with port 0.
         /// </summary>
         public DefaultSocketWrapper() :
             this(new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
         {
-            _innerSocket.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+            string dotnetBackendIPAddress = SparkEnvironment.ConfigurationService.GetBackendIPAddress();
+            if (dotnetBackendIPAddress == "localhost" || dotnetBackendIPAddress == null)
+            {
+                _innerSocket.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+            }
+            else
+            {
+                _innerSocket.Bind(new IPEndPoint(IPAddress.Parse(dotnetBackendIPAddress), 0));
```

Comment:
Hmm, shouldn't this be set to `IPAddress.Any` and dotnet backend ip address is passed to `connect`, no?